### PR TITLE
Add CBMC proof for MQTT_GetSubAckStatusCodes

### PR DESCRIPTION
**Pull request by aggarw13**
* Add CBMC proof for the `MQTT_GetSubAckStatusCodes` API function
* Update `MQTT_GetSubAckStatusCodes` API signature and its tests for issue found with the CBMC proof

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved. */
```
Comment (sarenameas) [inline libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c:1]: Can you please add the same copyright as the other harnesses

File: libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/Makefile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
```
Comment (sarenameas) [inline libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/Makefile:2]: Can you please copy the license from another proof's makefile

File: libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/Makefile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+INCLUDES +=
+
+REMOVE_FUNCTION_BODY +=
+UNWINDSET +=
+
+PROOF_SOURCES += $(PROOFDIR)/$(HARNESS_FILE).c
+PROOF_SOURCES += $(SRCDIR)/libraries/standard/mqtt/cbmc/sources/mqtt_cbmc_state.c
+PROJECT_SOURCES += $(SRCDIR)/libraries/standard/mqtt/src/mqtt.c
+
+include ../Makefile.common
```
Comment (sarenameas) [inline libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/Makefile:17]: Unfortunately there is some investigated bug that doesn't copy the newlines at the end of files. Could you please add the new lines?

File: libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * @file MQTT_GetSubAckStatusCodes_harness.c
+ * @brief Implements the proof harness for MQTT_GetSubAckStatusCodes function.
+ */
+
+#include "mqtt.h"
+#include "mqtt_cbmc_state.h"
+
+void harness()
```
Comment (sarenameas) [inline libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c:16]: Looks good!

File: libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/Makefile (hunk 3 above)
Comment (aggarw13) [reply libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/Makefile]: Added.

File: libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/Makefile (hunk 2 above)
Comment (aggarw13) [reply libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/Makefile]: Added

File: libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c (hunk 1 above)
Comment (aggarw13) [reply libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c]: Added

File: libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+{
+    MQTTPacketInfo_t * pSubackPacket;
+    uint8_t ** pPayloadStart;
+    size_t * pPayloadSize;
+
+    pSubackPacket = allocateMqttPacketInfo( NULL );
+    __CPROVER_assume( isValidMqttPacketInfo( pSubackPacket ) );
+
+    pPayloadStart = safeMalloc( sizeof( uint8_t * ) );
```
Comment (sarenameas) [inline libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c:39]: mallocCanFaile()

File: libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    pPayloadSize = safeMalloc( sizeof( uint8_t ) );
```
Comment (sarenameas) [inline libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c:40]: sizeof( size_t )

File: libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c (hunk 5 above)
Comment (aggarw13) [reply libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c]: Fixed

File: libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+{
+    MQTTPacketInfo_t * pSubackPacket;
+    uint8_t ** pPayloadStart;
+    size_t * pPayloadSize;
+
+    pSubackPacket = allocateMqttPacketInfo( NULL );
+    __CPROVER_assume( isValidMqttPacketInfo( pSubackPacket ) );
+
+    pPayloadStart = mallocCanFail( sizeof( uint8_t * ) );
+    pPayloadSize = mallocCanFail( sizeof( size_t ) );
```
Comment (markrtuttle) [inline libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c:40]: No assumptions required?  Not even that pPayloadStart points into pSubackPacket?

Comment (aggarw13) [reply libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c:40]: The `pPayloadStart` is an output parameter that will be pointing into `pSubackPacket` as the _outcome_ of the API call. Are you suggesting that the assumption could be added after the API call?

Comment (markrtuttle) [reply libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c:40]: My mistake.  Maybe add a comment that these are outputs?

Comment (aggarw13) [reply libraries/standard/mqtt/cbmc/proofs/MQTT_GetSubAckStatusCodes/MQTT_GetSubAckStatusCodes_harness.c:42]: Added

**Pull request merged**